eth_p2p/rlpx.nim

```diff
@@ -203,8 +203,13 @@ proc requestResolver[MsgType](msg: pointer, future: FutureBase) =
       else:
         doAssert false, "trying to resolve a timed out request with a value"
     else:
-      if not f.read.isSome:
-        doAssert false, "a request timed out twice"
+      try:
+        if not f.read.isSome:
+          doAssert false, "a request timed out twice"
+      except:
```

Comment:
Can you explain the reasoning here? The previous `doAssert` is put there to catch a potential bug in the code. Did it trigger for you?